### PR TITLE
fmtowns: bring machine configurations a bit closer to real hardware

### PR DESCRIPTION
**Pull request by rzero9**
This PR tries to better approximate what actual FM Towns machines are like, in terms of devices/configurations. The details are in the commit text, but the most important changes are the removal of the integrated SCSI controller on the first-gen models and the Marty (they don't have one in real life), and RAM options that reflect the actual onboard/SIMM combinations for each model. It also includes a couple of bugfixes and sanity checks to prevent emulator crashes I've found along the way.

This is the first time I mess around with devices and machine configurations, so feel free to criticize or suggest improvements.

Sources used:

http://www.mahoroba.ne.jp/~kazstary_/usage/simmfaq.txt
http://labo.main.jp/newage/towns/fmt_model.html
https://assemblergames.com/threads/fm-towns-marty-non-pcmcia-ram-expansion.65594/

As an aside, I may be able to acquire a Model 2 computer in the next few weeks/months and dump the ROMs, which would allow us to finally have actual emulation of the first-gen models from 1989, but I can't guarantee anything yet.

### CODE REVIEW

File: src/mame/drivers/fmtowns.cpp [hunk 1]
```diff
@@ -492,13 +492,15 @@ READ8_MEMBER(towns_state::towns_floppy_r)
...
+					if(m_flop[0]->get_device()->exists())
+						ret |= 0x03;
 				break;
 			case 2:
 				ret |= 0x0c;
-				if(m_flop[1]->get_device()->exists())
-					ret |= 0x03;
+				if(m_flop[1]->get_device())
+					if(m_flop[1]->get_device()->exists())
+						ret |= 0x03;
```
Comment (cuavas) [inline src/mame/drivers/fmtowns.cpp]: The `&&` operator has short-circuit semantics.

File: src/mame/includes/fmtowns.h [hunk 2]
```diff
@@ -138,36 +138,46 @@ class towns_state : public driver_device
...
+	void townsmx(machine_config &config);
 	void townssj(machine_config &config);
 
+	DECLARE_WRITE_LINE_MEMBER(towns_scsi_irq);
+	DECLARE_WRITE_LINE_MEMBER(towns_scsi_drq);
 	INTERRUPT_GEN_MEMBER(towns_vsync_irq);
-
+	DECLARE_READ16_MEMBER(towns_scsi_dma_r);
+	DECLARE_WRITE16_MEMBER(towns_scsi_dma_w);
+
```
Comment (cuavas) [inline src/mame/includes/fmtowns.h]: Why have you moved these to the public section?  Aren't they only accessed from machine configuration member functions?

Comment (rzero9) [reply src/mame/includes/fmtowns.h]: I meant to move them to protected after testing some ideas, and use the derived class on the machine configuration, but I never actually did it. Fixed.

File: src/mame/drivers/fmtowns.cpp (hunk 1 above)
Comment (rzero9) [reply src/mame/drivers/fmtowns.cpp]: Thanks, changed them to a single if statement.

**Pull request merged**